src/ast/scopes.cc

```diff
@@ -1183,7 +1183,7 @@ Variable* Scope::DeclareVariable(
   // immediately obvious how to do that at this level.
   // (RUN-2604)
   if (recordreplay::IsReplaying() &&
-      !recordreplay::FeatureEnabled("optimize-away") &&
+      recordreplay::FeatureEnabled("optimize-away") &&
```

Comment:
approved, but if `FeatureEnabled` is default on doesn't forcing allocation make sure the value is available?

Reply:
ah yeah, I should rename the flag

Reply:
done.  `optimize-away` is dead.  long live `force-variable-context-allocation`